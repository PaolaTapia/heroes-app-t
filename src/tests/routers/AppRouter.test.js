import { mount } from 'enzyme';
import React from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../routers/AppRouter';


describe('pruebas en AppRouter', () => {
const contextValue={
      dispatch: jest.fn(), 
      user: {
           logged: false
      }
}

     test('debe de mostrar el login si no está autenticado ', () => {
          const wrapper= mount(
               <AuthContext.Provider value={contextValue}>
                    <AppRouter/>
               </AuthContext.Provider>
          ); 

          expect(wrapper).toMatchSnapshot(); 
     }); 
     test('debe de mostrar el componente marvel si está autenticado', () => {
          const contextValue={
               dispatch: jest.fn(), 
               user: {
                    logged: true, 
                    name: 'Paola'
               }
         }


          const wrapper= mount(
               <AuthContext.Provider value={contextValue}>
                    <AppRouter/>
               </AuthContext.Provider>
          ); 
          
          expect(wrapper.find('.navbar').exists()).toBe(true); 
     }); 
     
     
})
