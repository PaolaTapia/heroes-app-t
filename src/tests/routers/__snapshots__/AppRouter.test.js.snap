// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en AppRouter debe de mostrar el login si no est√° autenticado  1`] = `
<div>
  <div
    className="container"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
